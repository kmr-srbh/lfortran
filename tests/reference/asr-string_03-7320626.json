{
    "basename": "asr-string_03-7320626",
    "cmd": "lfortran --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/string_03.f90",
    "infile_hash": "7815b8703dfd3a1c61cab40604e20693508878e8a90e450834ce592b",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-string_03-7320626.stdout",
    "stdout_hash": "e152110c6c26c12378e0da5c71862d824ca30fc1c9393f042ee174bb",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}