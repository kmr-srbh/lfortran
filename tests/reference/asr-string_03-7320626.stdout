(TranslationUnit
    (SymbolTable
        1
        {
            string_03:
                (Program
                    (SymbolTable
                        3
                        {
                            combined:
                                (Variable
                                    3
                                    combined
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Character 1 29 () PointerString)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            intro:
                                (ExternalSymbol
                                    3
                                    intro
                                    2 intro
                                    string_03_mod
                                    []
                                    intro
                                    Public
                                ),
                            last_name:
                                (Variable
                                    3
                                    last_name
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Character 1 7 () PointerString)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            posit:
                                (Variable
                                    3
                                    posit
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Character 1 5 () PointerString)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            title:
                                (Variable
                                    3
                                    title
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Character 1 4 () PointerString)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            verb:
                                (Variable
                                    3
                                    verb
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Character 1 8 () PointerString)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    string_03
                    [string_03_mod]
                    [(Assignment
                        (Var 3 verb)
                        (StringConstant
                            "learned "
                            (Character 1 8 () PointerString)
                        )
                        ()
                    )
                    (Assignment
                        (Var 3 posit)
                        (StringConstant
                            "from "
                            (Character 1 5 () PointerString)
                        )
                        ()
                    )
                    (Assignment
                        (Var 3 title)
                        (StringConstant
                            "the "
                            (Character 1 4 () PointerString)
                        )
                        ()
                    )
                    (Assignment
                        (Var 3 last_name)
                        (StringConstant
                            "best"
                            (Character 1 4 () PointerString)
                        )
                        ()
                    )
                    (Assignment
                        (Var 3 combined)
                        (StringConcat
                            (StringConcat
                                (StringConcat
                                    (StringConcat
                                        (StringConcat
                                            (Var 3 intro)
                                            (Var 3 verb)
                                            (Character 1 13 () PointerString)
                                            ()
                                        )
                                        (Var 3 posit)
                                        (Character 1 18 () PointerString)
                                        ()
                                    )
                                    (Var 3 title)
                                    (Character 1 22 () PointerString)
                                    ()
                                )
                                (Var 3 last_name)
                                (Character 1 29 () PointerString)
                                ()
                            )
                            (StringConstant
                                "."
                                (Character 1 1 () PointerString)
                            )
                            (Character 1 30 () PointerString)
                            ()
                        )
                        ()
                    )
                    (Print
                        [(Var 3 combined)]
                        ()
                        ()
                    )]
                ),
            string_03_mod:
                (Module
                    (SymbolTable
                        2
                        {
                            intro:
                                (Variable
                                    2
                                    intro
                                    []
                                    Local
                                    (StringConstant
                                        "I've "
                                        (Character 1 5 () PointerString)
                                    )
                                    (StringConstant
                                        "I've "
                                        (Character 1 5 () PointerString)
                                    )
                                    Parameter
                                    (Character 1 5 () PointerString)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    string_03_mod
                    []
                    .false.
                    .false.
                )
        })
    []
)
